src/cryptonote_core/oxen_name_system.cpp

```diff
@@ -100,26 +103,27 @@ std::pair<std::basic_string_view<unsigned char>, std::basic_string_view<unsigned
   return result;
 }
 
-std::string lns::mapping_value::to_readable_value(cryptonote::network_type nettype, lns::mapping_type type) const
+std::string ons::mapping_value::to_readable_value(cryptonote::network_type nettype, ons::mapping_type type) const
 {
   std::string result;
   if (is_lokinet_type(type))
   {
     result = oxenmq::to_base32z(to_view()) + ".loki";
-  }
-  else if (type == lns::mapping_type::wallet)
-  {
-    cryptonote::address_parse_info addr_info = {};
-    if (len == sizeof(addr_info))
-    {
-      std::memcpy(&addr_info, buffer.data(), len);
-      result = cryptonote::get_account_address_as_str(nettype, addr_info.is_subaddress, addr_info.address);
+  } else if (type == ons::mapping_type::wallet) {
+    cryptonote::address_parse_info addr_info{0};
+    auto iter = std::next(buffer.begin(),1);
+    std::memcpy(&addr_info.address.m_spend_public_key.data, &*iter, 32);
+    std::advance(iter,32);
+    std::memcpy(&addr_info.address.m_view_public_key.data, &*iter, 32);
+    if (buffer[0] == 0x2) {
+      std::advance(iter,32);
+      std::copy_n(iter,8,addr_info.payment_id.data);
+      addr_info.has_payment_id = true;
+    } else if (buffer[0] == 0x1) {
```

Comment:
I am guessing from the context that `0x1`/`0x2` values mean integrated addr/subaddress, but it'd be clearer if they were something like (though likely there are better name choices):
```C++
constexpr char ONS_WALLET_CODE_PRIMARY = 0x00;
constexpr char ONS_WALLET_CODE_INTEGRATED = 0x01;
constexpr char ONS_WALLET_CODE_SUBADDRESS = 0x02;
```